MirSurface: update visibility when surface's hidden is chagned

peat-psuwit opened the pull request:
Mir determines surface's visibility from both `has_submitted_buffer` and
`hidden` state. Thus, QtMir has to update its visibility whenever
surface's hidden is set.

The reason it previously worked most of the time is because surface's
hidden is synchronized with window's state and QtMir update its
visibility when window's state changed. However, this synchronization
happens in the Mir IPC thread, creating a race between this
synchroniztion and QtMir's updateVisibility() in the main thread.

mardy commented on src/modules/Unity/Application/mirsurface.cpp:
```diff
@@ -211,6 +211,10 @@ MirSurface::MirSurface(NewWindow newWindowInfo,
     connect(m_surfaceObserver.get(), &SurfaceObserver::attributeChanged, this, &MirSurface::onAttributeChanged);
     connect(m_surfaceObserver.get(), &SurfaceObserver::nameChanged, this, &MirSurface::onNameChanged);
     connect(m_surfaceObserver.get(), &SurfaceObserver::cursorChanged, this, &MirSurface::setCursor);
+    connect(m_surfaceObserver.get(), &SurfaceObserver::hiddenChanged, this, [&](bool) {
```
You should be able to connect to `&MirSurface::updateVisible`

UniversalSuperBox commented on src/modules/Unity/Application/mirsurface.cpp:
```diff
@@ -211,6 +211,10 @@ MirSurface::MirSurface(NewWindow newWindowInfo,
     connect(m_surfaceObserver.get(), &SurfaceObserver::nameChanged, this, &MirSurface::onNameChanged);
     connect(m_surfaceObserver.get(), &SurfaceObserver::cursorChanged, this, &MirSurface::setCursor);
+    connect(m_surfaceObserver.get(), &SurfaceObserver::hiddenChanged, this, [&](bool) {
+        // Mir determines visibility from it's hidden state.
```
"its" is possessive, "it's" is "it is"

peat-psuwit replied:
Well, I evaded this by removing it altogether. :D

peat-psuwit committed:
MirSurface: update visibility when surface's hidden is changed

Mir determines surface's visibility from both `has_submitted_buffer` and
`hidden` state. Thus, QtMir has to update its visibility whenever
surface's hidden is set.

The reason it previously worked most of the time is that surface's
hidden is synchronized with window's state and QtMir update its
visibility when window's state changed. However, this synchronization
happens in the Mir IPC thread, creating a race between this
synchronization and QtMir's updateVisibility() in the main thread.

UniversalSuperBox reviewed (approved):
After looking like an idiot dismissing and opening the keyboard for five minutes straight (four times today), I'm pretty sure this PR fixes the listed issue without side effects.
